vectordraw/grader.py

```diff
@@ -0,0 +1,345 @@
+"""
+This module contains grading logic for Vector Drawing exercises.
+"""
+
+# pylint: disable=invalid-name
+
+import inspect
+import logging
+import math
+
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+# Built-in check functions
+
+def _errmsg(default_message, check, vectors):
+    """
+    Return error message for `check` targeting a vector from `vectors`.
+
+    If `check` does not define a custom error message, fall back on `default_message`.
+    """
```

Comment:
It's pretty clear there are coordinated data structures in wide use throughout this module.  If there were a description of those structures at the top of the module, or a pointer to a description, it would make it easier to review, understand, and maintain this code.